### PR TITLE
[PyOpenSci] Add make_criteria helper

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR implements a suggestion made in #1335 
- [x] Tests for the changes have been added (for bug fixes / features)
  - [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] CHANGES.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Add a `make_criteria` helper to reshape dataset into 2D critera arrays, as expected by the ensemble reduction functions.

EDIT: Also, made this function a bit complicated so that it accepts dataset with variables of different shapes and still preserves the coordinates.

### Does this PR introduce a breaking change?
No.

### Other information:
~I need to add a test and udpate the history.~

### CODE REVIEW

**Review by tlogan2000 — APPROVED** [review]

File: xclim/ensembles/_reduce.py [hunk 1]
```diff
@@ -27,6 +27,30 @@
...
+    Returns
+    -------
+    crit : DataArray
+      Same data, reshaped.
+    """
+    if isinstance(ds, xarray.Dataset):
+        ds = ds.to_array("variables")
+    return ds.stack(criteria=set(ds.dims) - {"realization"}).rename("criteria")
+
+
```
[the remark this answers is not in the document]
Comment (aulemahal) [reply xclim/ensembles/_reduce.py]: Oups, forgot to address this. It's about the indentation, innit?

File: xclim/ensembles/_reduce.py [hunk 2]
```diff
@@ -27,6 +28,85 @@
...
+    -------
+    crit : DataArray
+        Same data, reshaped. Old coordinates (and variables) are available as a multiindex.
+
+    Notes
+    -----
+    One can get back to the original dataset with
+
+    >>> crit = make_criteria(ds)
+    >>> ds2 = crit.unstack("criteria").to_dataset("variables")
```
Comment (Zeitsperre) [inline xclim/ensembles/_reduce.py:55]: The reason for build failures is that this is actually running! If we want, you can replace `ds` with one of the existing dataset aliases: https://github.com/Ouranosinc/xclim/blob/78ff493ded25907c4e17f0695950d1e04ff88d28/xclim/testing/tests/data.py#L108

Comment (aulemahal) [reply xclim/ensembles/_reduce.py:55]: Dang, I thought that placing it in `notes` (and not `examples`) would make it _not_ run.

Comment (aulemahal) [reply xclim/ensembles/_reduce.py]: All "datasets" in the namespace are actually DataArrays... I changed to a code-block instead. (this shouldn't run, right?)

Comment (Zeitsperre) [reply xclim/ensembles/_reduce.py]: Correct!

**Pull request merged**